File: backend/semgrep_rules/tests/javascript/kisa_sw_08_xxe.js
const libxmljs = require("libxmljs");

function unsafe(xml) {
  // ruleid: kisa.sw08.javascript.unsafe-xml-external-entity
  return libxmljs.parseXml(xml, { noent: true });
}

function safe(xml) {
  // ok: kisa.sw08.javascript.unsafe-xml-external-entity
  return libxmljs.parseXml(xml, { noent: false });
}

File: backend/semgrep_rules/tests/python/kisa_sw_08_xxe.py
from lxml import etree

def unsafe():
    # ruleid: kisa.sw08.python.unsafe-xml-external-entity
    return etree.XMLParser(resolve_entities=True)

def safe():
    # ok: kisa.sw08.python.unsafe-xml-external-entity
    return etree.XMLParser(resolve_entities=False, load_dtd=False, no_network=True)
